File: app/src/main/java/com/neworld/youyou/activity/SplashActivity.java
package com.neworld.youyou.activity;

import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.neworld.youyou.MainActivity;
import com.neworld.youyou.R;
import com.neworld.youyou.utils.Sputil;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        initTime();
    }

    private void initTime() {
        new Thread(new Runnable() {
            @Override
            public void run() {
                try {
                    Thread.sleep(2000);
                    initUsr();
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
        }).start();
    }

    private void initUsr() {
        String userId = Sputil.getString(SplashActivity.this, "userId", "");
        if (userId != null && userId.length() > 0) {
            startActivity(new Intent(SplashActivity.this, MainActivity.class));
        } else {
            startActivity(new Intent(SplashActivity.this, LoginActivity.class));
        }
        finish();
    }

    /** * 判断当前网络是否wifi */
    public boolean isWifi(Context mContext) {
        ConnectivityManager connectivityManager = (ConnectivityManager) mContext .getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetInfo = connectivityManager.getActiveNetworkInfo();
        if (activeNetInfo != null && activeNetInfo.getType() == ConnectivityManager.TYPE_WIFI) {
            return true;
        }
        return false;
    }

}

File: app/src/main/java/com/neworld/youyou/activity/SplashActivity.kt
package com.neworld.youyou.activity

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.os.Handler
import android.support.v4.app.ActivityCompat
import android.support.v4.content.ContextCompat
import android.widget.Toast
import com.neworld.youyou.MainActivity
import com.neworld.youyou.R
import com.neworld.youyou.add.base.Activity
import com.neworld.youyou.utils.preference
import com.neworld.youyou.utils.showToast
import kotlinx.android.synthetic.main.item_parent.*

/**
 * @author by user on 2018/3/5.
 */
class SplashActivity : Activity() {

    private val userId by preference("userId", "")

    override fun getContentLayoutId() = R.layout.activity_splash

    override fun initWidget() {
        if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
                != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE,
                    Manifest.permission.READ_EXTERNAL_STORAGE),
                    2)
        } else {
            toHome()
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        if (grantResults.isEmpty() || grantResults[0] == PackageManager.PERMISSION_DENIED ||
                grantResults[1] == PackageManager.PERMISSION_DENIED) {
            Toast.makeText(this, "无读写权限可能会导致部分功能不可用或报错，请尽快到软件管理处添加权限。",
                    Toast.LENGTH_LONG).show()
            Handler().postDelayed({
                finish()
            }, 2000)
        } else {
            toHome()
        }
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }

    private fun toHome() {
        Handler().postDelayed({
            if (userId.isEmpty()) {
                startActivity(Intent(this@SplashActivity, LoginActivity::class.java))
            } else {
                startActivity(Intent(this@SplashActivity, MainActivity::class.java))
            }
            finish()
        }, 2000)
    }
}